### ISSUE TITLE
(aws-apigatewayv2-integrations): How to deploy the LambdaProxyIntegration breaking change

### ISSUE THREAD

**Issue by lletourn** [aws/aws-cdk#18693]
### General Issue

Issue with version upgrade deployment

### The Question

From v1.134.0 to v1.135.0 a documented breaking change got introduced. (git changeset 29039e8bd13a4fdb7f84254038b3331c179273fd)

I changed my cdk code from
`http_api = apigatewayv2.HttpApi(
            self,
            'authenticationApi',
            api_name='AuthenticationApi',
            default_integration=apigatewayv2_integrations.LambdaProxyIntegration(handler=authentication_lambda),
            default_domain_mapping={
                'domain_name': dn
            }
        )`
to
`http_api = apigatewayv2.HttpApi(
            self,
            'authenticationApi',
            api_name='AuthenticationApi',
            default_integration=aws_cdk.aws_apigatewayv2_integrations.HttpLambdaIntegration("authenticationLambdaIntegration", handler=authentication_lambda),
            default_domain_mapping={
                'domain_name': dn
            }
        )
`

But now I can't deploy because I get an excecption
`10:15:23 a.m. | UPDATE_FAILED        | AWS::ApiGatewayV2::Route       | authenticationApiDefaultRouteDB2326C1
Invalid route identifier specified (Service: AmazonApiGatewayV2; Status Code: 404; Error Code: NotFoundException; Request ID: 33887451-859f-478a-b47a-eb4f859fbc11; Proxy: null)
`
How can we update our cloudformation with the new id from the old without deleting everything in production?

Thanks

### CDK CLI Version

v1.135.0

### Node.js Version

v16.13.2

### OS

ubuntu 20.04

### Language

Python

### Language Version

3.8.10

**Comment by lletourn** [comment]
I tried to reproduce and couldn't in another AWS account I have.
Maybe my issue is something else in that account...

**Comment by lletourn** [comment]
Better exception:

`
SsoStackVal |  8/28 | 3:43:08 p.m. | UPDATE_FAILED        | AWS::ApiGatewayV2::Route       | authenticationApi/DefaultRoute (authenticationApiDefaultRouteDB2326C1) Invalid route identifier specified (Service: AmazonApiGatewayV2; Status Code: 
404; Error Code: NotFoundException; Request ID: 5027b992-a71d-42ed-8f27-fbbeee25c9bb; Proxy: null)

	new HttpRoute (/tmp/jsii-kernel-fnvTwI/node_modules/@aws-cdk/aws-apigatewayv2/lib/http/route.js:124:23)
	\_ new HttpApi (/tmp/jsii-kernel-fnvTwI/node_modules/@aws-cdk/aws-apigatewayv2/lib/http/api.js:145:13)
	\_ /tmp/tmpdmp_7h_s/lib/program.js:8432:58
	\_ Kernel._wrapSandboxCode (/tmp/tmpdmp_7h_s/lib/program.js:8860:24)
	\_ Kernel._create (/tmp/tmpdmp_7h_s/lib/program.js:8432:34)
	\_ Kernel.create (/tmp/tmpdmp_7h_s/lib/program.js:8173:29)
	\_ KernelHost.processRequest (/tmp/tmpdmp_7h_s/lib/program.js:9757:36)
	\_ KernelHost.run (/tmp/tmpdmp_7h_s/lib/program.js:9720:22)
	\_ Immediate._onImmediate (/tmp/tmpdmp_7h_s/lib/program.js:9721:46)
	\_ processImmediate (node:internal/timers:464:21)
`

**Comment by peterwoodworth** [comment]
Hey @lletourn, using a new construct will create a new logical ID (and will thus cause a recreation of your resources). Here's a code snippet which will allow you to override the logical ID of the generated integration

```ts
    const myApi = new HttpApi(this, 'api', {
      apiName: 'ApiName',
      // defaultIntegration: new LambdaProxyIntegration({
      //   handler: fn
      // }),
      defaultIntegration: new HttpLambdaIntegration('MyIntegration', fn),
      defaultDomainMapping: {
        domainName: DomainName.fromDomainNameAttributes(this, 'DomainName', {
          name: 'Name',
          regionalDomainName: 'regionalDomainName',
          regionalHostedZoneId: 'RegionalID'
        })
      }
    });

    const defaultRoute = myApi.node.findChild('DefaultRoute');
    const myInt = defaultRoute.node.findChild('MyIntegration');
    const resource = myInt.node.findChild('Resource') as CfnIntegration;
    resource.overrideLogicalId('apiDefaultRouteHttpIntegrationc467df7af29c2ba9300e2d0928438acbAE00E844');
```

**Comment by lletourn** [comment]
Feels a bit dirty, but works great. Sad my production code will need these lines foreever (or until I find some downtime to recreate the api gateway part).
thanks!

**Comment by peterwoodworth** [comment]
Yeah, unfortunately there's no pretty way to override the logical id of a child construct. Not the hugest deal though, since this should remain consistent 🙂
